### PR TITLE
tests/shell: fix flake8 errors in Python script

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR cleans up the Python automated test script of the `tests/shell` application: there are a couple of errors because of escape characters used in test strings. One can be fixed by using a raw string, the second one is ignored because I don't know how to fix it :)

```
tests/sys/shell/tests/01-run.py:27:10: W605 invalid escape sequence '\d'
tests/sys/shell/tests/01-run.py:77:16: W605 invalid escape sequence '\e'
tests/sys/shell/tests/01-run.py:77:36: W605 invalid escape sequence '\ '
```

### Testing procedure

- A green CI
- `tests/shell` test is still working:

<details>

```
$ make -C tests/shell all test --no-print-directory 
Building application "tests_shell" for "native" with MCU "native".

"make" -C /work/riot/RIOT/boards/native
"make" -C /work/riot/RIOT/boards/native/drivers
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/native
"make" -C /work/riot/RIOT/cpu/native/periph
"make" -C /work/riot/RIOT/cpu/native/stdio_native
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/app_metadata
"make" -C /work/riot/RIOT/sys/auto_init
"make" -C /work/riot/RIOT/sys/ps
"make" -C /work/riot/RIOT/sys/shell
"make" -C /work/riot/RIOT/sys/shell/commands
"make" -C /work/riot/RIOT/sys/test_utils/interactive_sync
/usr/bin/ld: /work/riot/RIOT/tests/shell/bin/native/cpu/tramp.o: warning: relocation against `_native_saved_eip' in read-only section `.text'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
   text	   data	    bss	    dec	    hex	filename
  30253	    812	  47812	  78877	  1341d	/work/riot/RIOT/tests/shell/bin/native/tests_shell.elf


/work/riot/RIOT/tests/shell/bin/native/tests_shell.elf  
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2021.01-devel-573-gcf154a-pr/tests/shell_script_enh)
test_shell.
> 
> 
> bufsize
bufsize
128
> ________________________________________________________________________________________________________________________________verylong
________________________________________________________________________________________________________________________________verylong
shell: maximum line length exceeded
> garbage1234
garbage1234
> 
> echo____________________________________________________________________________________________________________________________verylong
_________echo                                                                                                                           
"echo"
> 
shell exited
> 
> start_test
start_test
[TEST_START]
> 


> 
> echo a string
echo a string
"echo""a""string"
> echo   multiple   spaces   between   argv
echo   multiple   spaces   between   argv
"echo""multiple""spaces""between""argv"
> echo 	 tabs		 processed 		like	 		spaces
echo 	 tabs		 processed 		like	 		spaces
"echo""tabs""processed""like""spaces"
> unknown_command
unknown_command
shell: command not found: unknown_command
>      echo leading spaces
     echo leading spaces
"echo""leading""spaces"
> 					echo leading tabs
					echo leading tabs
"echo""leading""tabs"
> echo trailing spaces     
echo trailing spaces     
"echo""trailing""spaces"
> echo trailing tabs					
echo trailing tabs					
"echo""trailing""tabs"
> hello-world
hello-world
shell: command not found: hello-world
echo
echo
"echo"
> echo \'
echo \'
"echo""'"
> echo \"
echo \"
"echo""""
> echo escaped\ space
echo escaped\ space
"echo""escaped space"
> echo escape within '\s\i\n\g\l\e\q\u\o\t\e'
echo escape within '\s\i\n\g\l\e\q\u\o\t\e'
"echo""escape""within""singlequote"
> echo escape within "\d\o\u\b\l\e\q\u\o\t\e"
echo escape within "\d\o\u\b\l\e\q\u\o\t\e"
"echo""escape""within""doublequote"
> echo "t\e st" "\"" '\'' a\ b
echo "t\e st" "\"" '\'' a\ b
"echo""te st"""""'""a b"
> echo "hello"world
echo "hello"world
"echo""helloworld"
> echo hel"lowo"rld
echo hel"lowo"rld
"echo""helloworld"
> echo hello"world"
echo hello"world"
"echo""helloworld"
> echo quoted space " "
echo quoted space " "
"echo""quoted""space"" "
> echo abc"def'ghijk"lmn
echo abc"def'ghijk"lmn
"echo""abcdef'ghijklmn"
> echo abc'def"ghijk'lmn
echo abc'def"ghijk'lmn
"echo""abcdef"ghijklmn"
> echo "'" '"'
echo "'" '"'
"echo""'""""
> echo a\
echo a\
shell: incorrect quoting
> echo "
echo "
shell: incorrect quoting
> echo '
echo '
shell: incorrect quoting
> echo abcdef"ghijklmn
echo abcdef"ghijklmn
shell: incorrect quoting
> echo abcdef'ghijklmn
echo abcdef'ghijklmn
shell: incorrect quoting
> ps
ps
	pid | state    Q | pri 
	  1 | pending  Q |  15
	  2 | running  Q |   7
> help
help
Command              Description
---------------------------------------
bufsize              Get the shell's buffer size
start_test           starts a test
end_test             ends a test
echo                 prints the input command
reboot               Reboot the node
version              Prints current RIOT_VERSION
pm                   interact with layered PM subsystem
ps                   Prints information about running threads.
app_metadata         Returns application metadata
> reboot
reboot


		!! REBOOT !!

RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2021.01-devel-573-gcf154a-pr/tests/shell_script_enh)
test_shell.
> end_test
end_test
[TEST_END]
> 

> 
> 
```

</details>

### Issues/PRs references

Raised when running the static test locally in #15358

### CODE REVIEW

**Comment by aabadie** [comment]
arg, wrong PR :)

File: tests/shell/tests/01-run.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
     ('echo escaped\\ space', '"echo""escaped space"'),
     ('echo escape within \'\\s\\i\\n\\g\\l\\e\\q\\u\\o\\t\\e\'', '"echo""escape""within""singlequote"'),
     ('echo escape within "\\d\\o\\u\\b\\l\\e\\q\\u\\o\\t\\e"', '"echo""escape""within""doublequote"'),
-    ("""echo "t\e st" "\\"" '\\'' a\ b""", '"echo""te st"""""\'""a b"'),
+    ("""echo "t\e st" "\\"" '\\'' a\ b""", '"echo""te st"""""\'""a b"'),  # noqa: W605
```
Comment (miri64) [inline tests/shell/tests/01-run.py:77]: Not sure I like suppressing linting on this very unreadable string. :-/

Comment (miri64) [reply tests/shell/tests/01-run.py:77]: Or rather: hard to parse string

Comment (fjmolinas) [reply tests/shell/tests/01-run.py:77]: Isn't the string supposed to be hard toparse?

Comment (miri64) [reply tests/shell/tests/01-run.py:77]: I mean by the human eye, not the shell tokenizer ^^"

Comment (fjmolinas) [reply tests/shell/tests/01-run.py:77]: So: `Invalid escape sequence 'x' (W605)` is the issue, can this be fixed by using raw strings and escape what is needed @aabadie ?

Comment (miri64) [reply tests/shell/tests/01-run.py:77]: You can't escape in raw strings, that's the point about them. Let me think about it.

Comment (miri64) [inline tests/shell/tests/01-run.py:77]:
```suggestion
    ("echo \"t\e st\" \"\\"\" '\\'' a\ b", '"echo""te st"""""\'""a b"'),
```
... not sure this is easier to read. So maybe suppressing the warning is the better way to go :sweat_smile:

Comment (miri64) [reply tests/shell/tests/01-run.py:77]: Github did make this its own review. See https://github.com/RIOT-OS/RIOT/pull/15373#pullrequestreview-524096821 why I resolved this conversation.

**Comment by aabadie** [comment]
So should we keep the PR as it is now ? (I'm a bit lost ;) )

**Comment by miri64** [comment]
> So should we keep the PR as it is now ? (I'm a bit lost ;) )

Yepp. Sorry for the confusion. Will test this PR now!

**Review by miri64 — APPROVED** [review]
ACK. I ran the test for `native`, `iotlab-m3` and `feather-m0` (because I had the later connected incidentally). It still succeeds for `native` and `iotlab-m3`, but fails on `feather-m0`, but that is already the case on master and probably because that board uses `usb-acm` for STDIO.
